lib/drivers/bayerContourNext.js

```diff
@@ -0,0 +1,501 @@
+    return {
+      packet: buildAckPacket(),
+      parser: packetParser
+    };
+  };
+
+    //header data looks like
+  /*
+  <STX>1H|\^&||qvqOi8|Bayer7350^01.14\01.03\04.18^7358-1611135^0000-
+  |A=1^C=00^G=es,en\es\it\de\fr\hr\da\nl\fi\el\no\pt\sl\sv^I=0200^R=
+  0^S=01^U=0^V=20600^X=070070070180130180070130^Y=120054252099^Z=1|4
+  |||||P|1|201505291248<ETB>01<CR><LF>
+  */
+
+  var parseHeader = function (s){
+    var data = s.split('\n').filter(function(e){ return e !== '';});
+    var header = data.shift();
+
+    if(verifyChecksum(header)){
+      var patient = data.shift();
+      var lineFeed = data.pop();
+      var pString = header.split('|');
+      var pInfo = pString[4].split('^');
+      var sNum = pInfo[2].match(/^\d+\-(\d+)/);
+      var records = data;
+      var threshold = pString[5].split('^')[7].match(/^.+\=(\d{2})(\d{3})/);
+
+      var devInfo = {
+        model: pInfo[0],
+        serialNumber: sNum[1],
+        nrecs: pString[6],
+        rawrecords: records,
+        lowT: parseInt(threshold[1]),
+        hiT: parseInt(threshold[2])
+      };
+
+      return devInfo;
+    }else{
+      return null;
+    }
+  };
+
+  function verifyChecksum(record){
+      var str = record.trim();
+      var data = str.split(String.fromCharCode(ASCII_CONTROL.ETB));
+      var check = data[1];
+      var sum = 0;
+      var n = record.slice(0, record.length - 3);
+
+      _.map(n, function(e){
+          if(e.charCodeAt(0) !== ASCII_CONTROL.STX){
+              sum += e.charCodeAt(0);
+          }
+      });
+
+      if((sum % 256) !== parseInt(check, 16)){
+          return null;
+      }else{
+          return data[0];
+      }
+  }
+
+  /* Record data looks like
+  <STX>5R|3|^^^Glucose|93|mg/dL^P||A/M0/T1||201505261150<CR><ETB>74<CR><LF>
+  */
+  var parseDataRecord = function (str, callback){
+    var data = verifyChecksum(str);
+    if(data){
+      var result = data.trim().match(/^.*\d+R\|(\d+).*Glucose\|(\d+)\|(\w+\/\w+).*\|{2}(.*)\|{2}(\d{12})$/).slice(1,6);
+      callback(null, result);
+    }else{
+      throw( new Error('Invalid record data') );
+    }
+
+  };
+
+  var getAnnotations = function (annotation, data){
+    if (annotation === '>') {
+      return [{
+        code: 'bg/out-of-range',
+        value: 'high',
+        threshold: data.hiT
+      }];
+    } else if (annotation === '<') {
+      return [{
+        code: 'bg/out-of-range',
+        value: 'low',
+        threshold: data.lowT
+      }];
+    } else {
+      return null;
+    }
+  };
+
+  var getOneRecord = function (record, data, callback) {
+    parseDataRecord(record,function(err,r){
+      var robj = {};
+      if (err) {
+          debug('Failure trying to read record', record);
+          debug(err);
+          return callback(err, null);
+      } else {
+          robj.timestamp = parseInt(r[4]);
+          robj.annotations = getAnnotations(r[3], data);
+          robj.units = r[2];
+          robj.glucose = parseInt(r[1]);
```

Comment:
mmm... I quite don't understand this and maybe is because we don't see the whole picture. The threshold values are been gathered from the header l:164, normally we get here two values ie. 20, 600, and according to the manual, this values represents GLU LO = 20 and GLU HI = 600.  So please could you elaborate a little more specially about 'threshold + 1'.